pkg/alertmanager/multitenant.go

```diff
@@ -737,6 +774,24 @@ func (am *MultitenantAlertmanager) ServeHTTP(w http.ResponseWriter, req *http.Re
 		return
 	}
 
+	if am.cfg.ShardingEnabled && am.distributor.IsPathSupported(req.URL.Path) {
+		am.distributor.DistributeRequest(w, req)
+		return
+	}
+
+	// If sharding is not enabled or Distributor does not support this path,
+	// it is served by this instance.
+	am.serveHTTPHelper(w, req)
+}
+
+// HandleRequest serves the Alertmanager's web UI and API sent via gRPC.
+func (am *MultitenantAlertmanager) HandleRequest(ctx context.Context, in *alertmanagerpb.Request) (*alertmanagerpb.Response, error) {
+	resp, err := am.grpcServer.Handle(ctx, in.HttpRequest)
+	return &alertmanagerpb.Response{HttpResponse: resp}, err
+}
+
+// serveHTTPHelper serves the Alertmanager's web UI and API.
+func (am *MultitenantAlertmanager) serveHTTPHelper(w http.ResponseWriter, req *http.Request) {
```

Comment:
It's not really a helper, but rather internal implementation of HTTP handler, without "sharding" check. Can we find a better name?